internal/pkg/action/operator_list_custom_resources.go

```diff
@@ -0,0 +1,113 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/api/pkg/operators/v2alpha1"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type CustomResourceLister interface {
+	List(ctx context.Context, ownedCRDS []v1alpha1.CRDDescription) (metav1.PartialObjectMetadataList, error)
+}
+
+type CustomResourceDefinitionFinder interface {
+	FindOperator(ctx context.Context, packageName, namespace string) (*v2alpha1.Operator, error)
+	Unzip(ctx context.Context, operator *v2alpha1.Operator) ([]v1alpha1.CRDDescription, error)
+}
+
+// OperatorListCustomResources knows how to find and list custom resources given a package name and namespace.
+type OperatorListCustomResources struct {
+	config  *Configuration
+	ListAll bool
+	PackageName string
+	Namespace string
+}
+
+func NewOperatorListCustomResources(cfg *Configuration) *OperatorListCustomResources {
+	return &OperatorListCustomResources{
+		config: cfg,
+		Namespace: cfg.Namespace,
+	}
+}
+
+// FindOperator finds an operator object on-cluster provided a package and namespace. Returns a copy of the operator.
+func (o *OperatorListCustomResources) FindOperator(ctx context.Context) (v2alpha1.Operator, error) {
+	opKey := types.NamespacedName{
+		Name:      o.PackageName,
+		Namespace: o.Namespace,
+	}
+	operator := v2alpha1.Operator{}
+	err := o.config.Client.Get(ctx, opKey, &operator)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return v2alpha1.Operator{}, fmt.Errorf("could not find operator for package %s in namespace %s", o.PackageName, o.Namespace)
```

Comment:
There's an error type that exists for this case. Perhaps we can reuse?
```suggestion
			return nil, ErrPackageNotFound{o.PackageName}
```